### PR TITLE
Fix typescript type errors in react admin `./client/two-column/tasks`

### PR DESCRIPTION
**Pull request by chihsuan**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Part of https://github.com/woocommerce/woocommerce/issues/32581

Fixes typescript type errors in react admin ./client/two-column/tasks

### How to test the changes in this Pull Request:

1. Run `pnpm nx build @woocommerce/data`
2. Run `pnpm tsc -b tsconfig.base.json | grep ./client/two-column/tasks` and confirm no errors.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?
* [x] Have you created a changelog file by running `pnpm nx affected --target=changelog`?

### FOR PR REVIEWER ONLY:

* [x] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: packages/js/data/src/onboarding/types.ts [hunk 1]
```diff
@@ -3,17 +3,23 @@ export type TaskType = {
...
 	isSnoozed: boolean;
 	isVisible: boolean;
-	isSnoozable: boolean;
+	isSnoozeable: boolean;
 	isDisabled: boolean;
 	snoozedUntil: number;
 	time: string;
 	title: string;
 	isVisited: boolean;
+	additionalInfo: string;
```
Comment (chihsuan) [inline packages/js/data/src/onboarding/types.ts:18]: reference:

https://github.com/woocommerce/woocommerce/blob/62de7bae9c715804b290c63ee279cdffccee6324/plugins/woocommerce/src/Admin/Features/OnboardingTasks/Task.php#L446-L466

File: packages/js/data/src/onboarding/types.ts [hunk 2]
```diff
@@ -27,13 +33,16 @@ export type TaskListSection = {
 
 export type TaskListType = {
```
Comment (chihsuan) [inline packages/js/data/src/onboarding/types.ts:34]: reference:
https://github.com/woocommerce/woocommerce/blob/62de7bae9c715804b290c63ee279cdffccee6324/plugins/woocommerce/src/Admin/Features/OnboardingTasks/TaskList.php#L403-L417

File: packages/js/data/src/user/use-user-preferences.ts [hunk 3]
```diff
@@ -2,20 +2,58 @@
...
+import schema, { Schema } from 'wordpress__core-data';
+import type { getEntityRecord } from 'wordpress__core-data/selectors';
 
 /**
  * Internal dependencies
  */
 import { STORE_NAME } from './constants';
 
+type UserPreferences = {
+	activity_panel_inbox_last_read?: string;
```
Comment (chihsuan) [inline packages/js/data/src/user/use-user-preferences.ts:19]: Screen Shot 2022-04-20 at 16 19 19

File: packages/js/data/src/user/use-user-preferences.ts [hunk 4]
```diff
@@ -2,20 +2,58 @@
...
+	products_report_columns?: string;
+	revenue_report_columns?: string;
+	task_list_tracked_started_tasks?: {
+		[ key: string ]: number;
+	};
+	taxes_report_columns?: string;
+	variations_report_columns?: string;
+};
+
+type WoocommerceMeta = UserPreferences & {
```
Comment (chihsuan) [inline packages/js/data/src/user/use-user-preferences.ts:42]: Screen Shot 2022-04-20 at 16 19 19

**Review by moon0326 — APPROVED** [review]
Nice work @chihsuan 

LGTM and tested well 🚀

**Pull request merged**